docs/src/cli/deploy-a-program.md

```diff
@@ -297,6 +298,59 @@ the program:
 solana program deploy --program-id <PROGRAM_ADDRESS> --buffer <BUFFER_ADDRESS>
 ```
 
-Note, the buffer's authority must match the program's upgrade authority.
+Note, the buffer's authority must match the program's upgrade authority. Also, upon successful deploy
+buffer accounts contents are copied into program accounts and are erased from blockchain.
 
 Buffers also support `show` and `dump` just like programs do.
+
+### Deploying program with offline signer authority
+
+Suggestion: the simplest way to do offline signing is to arrange your signers such that `--program-id`,
+`--buffer-authority` and `--upgrade-authority` correspond to offline signer (key-pair).
+
+The general process is as follows:
+1) (use online machine) create buffer
+2) (use online machine) upgrade buffer authority to offline signer
+3) (optional, use a separate online machine) verify the actual buffer on-chain contents
+4) (use offline machine) get a signature for your intent to create or upgrade program
+5) (use online machine) use this signature to build and broadcast create/upgrade transactions on-chain
+
+```bash
+# (1) (use online machine) create buffer
+solana program write-buffer <PROGRAM_FILEPATH>
+
+# (2) (use online machine) upgrade buffer authority to offline signer
+solana program set-buffer-authority <PUB_KEY> --new-buffer-authority <PUB_KEY>
+```
+
+- note, above we have to issue a separate `set-buffer-authority` command (using online machine) to change authority to
+  be that of offline signer because we don't have access to offline signer private key to sign up as buffer authority
+  when running `solana program write-buffer` command using online machine (offline signer private key is stored on
+  offline machine)
```

Comment:
Gotta say, this behavior seems a bit weird to me (not being able to specify the desired buffer authority from the get go), 

any reason for why this is the way it is (perhaps somehow can be fixed) ?